### PR TITLE
feat(deps)!: support for djs v14

### PR DESCRIPTION
**Pull request by vladfrangu**
Related:
- [framework#512](https://github.com/sapphiredev/framework/pull/512)
- [plugins#403](https://github.com/sapphiredev/plugins/pull/403)

---

This PR brings support for djs v14.

### CODE REVIEW

**Comment by vladfrangu** [comment]
@sapphiredev pack this 🙏

File: package.json [hunk 1]
```diff
@@ -19,20 +19,20 @@
...
 		"@types/ws": "^8.5.3",
-		"@typescript-eslint/eslint-plugin": "^5.33.1",
-		"@typescript-eslint/parser": "^5.33.1",
+		"@typescript-eslint/eslint-plugin": "^5.34.0",
+		"@typescript-eslint/parser": "^5.34.0",
 		"@vitest/coverage-c8": "^0.22.1",
 		"cz-conventional-changelog": "^3.3.0",
-		"discord-api-types": "^0.33.5",
-		"discord.js": "^13.10.2",
+		"discord-api-types": "^0.37.4",
```
Comment (imranbarbhuiya) [inline package.json:34]: dapi-types can be removed now as it's reexported from djs.

File: packages/discord.js-utilities/src/lib/type-guards.ts [hunk 2]
```diff
@@ -1,24 +1,24 @@
 import { isNullish, Nullish } from '@sapphire/utilities';
-import type { APIGuildMember, APIInteractionDataResolvedGuildMember, APIInteractionGuildMember, APIMessage } from 'discord-api-types/v9';
+import { APIGuildMember, APIInteractionDataResolvedGuildMember, APIInteractionGuildMember, APIMessage, ChannelType } from 'discord-api-types/v10';
 import {
+	BaseInteraction,
```
Comment (imranbarbhuiya) [inline packages/discord.js-utilities/src/lib/type-guards.ts:4]:
```suggestion
import { 
        type APIGuildMember,
        type APIInteractionDataResolvedGuildMember,
        type APIInteractionGuildMember,
        type APIMessage,
        ChannelType,
	BaseInteraction,
```

**Review by imranbarbhuiya — CHANGES_REQUESTED** [review]

File: packages/discord.js-utilities/src/lib/type-guards.ts (hunk 2 above)
Comment (favna) [reply packages/discord.js-utilities/src/lib/type-guards.ts:4]: And organized imports after 
(VSCode organizer imports puts types last)

File: package.json (hunk 1 above)
Comment (vladfrangu) [reply package.json]: Keeping it for editor autocompletes

**Comment by vladfrangu** [comment]
@sapphiredev pack

**Comment by favna** [comment]
@sapphiredev pack

**Comment by favna** [comment]
@sapphiredev pack

**Comment by favna** [comment]
@sapphiredev pack

**Comment by favna** [comment]
@sapphiredev pack

**Comment by favna** [comment]
@sapphiredev pack

**Pull request merged**